extensions/amp-story/1.0/amp-story-draggable-drawer.js

```diff
@@ -238,6 +240,27 @@ export class DraggableDrawer extends AMP.BaseElement {
     this.headerEl.toggleAttribute('desktop', !isMobile);
   }
 
+  /**
+   * Reacts to viewport size changes.
+   * @protected
+   */
+  onViewportResize_() {
+    const contentHeight = this.contentEl./*OK*/ getBoundingClientRect().height;
+    const viewportHeight = this.getViewport().getHeight();
+    if (contentHeight > viewportHeight) {
+      // Prevent the height of the draggable drawer from ever exceeding that of
+      // the viewport, ensuring that the drawer header is always visible. This
+      // is particularly necessary on Android because opening the soft keyboard
+      // triggers a viewport resize that we counteract by setting the story
+      // size.
+      this.mutateElement(() => {
+        setImportantStyles(this.element, {'height': px(viewportHeight)});
```

Comment:
Is this OK for this to run on a resize triggered by anything else than the keyboard opening?

Reply:
I think this should be OK to run on any type of resize because we generally should not want the draggable drawer's height to ever exceed the viewport's height, or else the header and close button would probably be hidden above the viewport.